code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/AccountAccess.java

```diff
@@ -0,0 +1,55 @@
+package io.cattle.platform.iaas.api.auth;
+
+import io.cattle.platform.api.auth.ExternalId;
+import io.cattle.platform.core.model.Account;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class AccountAccess {
+
+    Account account;
+    Account project;
+    Set<ExternalId> externalIds;
+
+    public AccountAccess() {
+    }
+
+    public AccountAccess(Account account, Set<ExternalId> externalIdSet) {
+        this.account = account;
+        this.project = account;
+        this.externalIds = externalIdSet;
+        if (account != null){
```

Comment:
No need to create extra HashSets that just adds more work for the gc and the null check on account is not necessary you are correct.